### ISSUE TITLE
Remote watching is repeated

### ISSUE THREAD

**Issue by rogeralsing** [akkadotnet/akka.net#1090]
If like 243 is uncommented in the RemoteSystemDaemon: 
```csharp
           //TODO: figure out why current transport is not set when this message is sent
          parent.Tell(new Watch(parent, this),this);
```

The system that trigger the watch will repeatedly send `Re-watching` to the remote system.
![remotewatch](https://cloud.githubusercontent.com/assets/647031/8315934/1b5c5250-19f4-11e5-9213-77a02f3c74d2.png)

I guess this is not expected(?)
What and why is the reason of a re-watch?

cc @Aaronontheweb

**Comment by Aaronontheweb** [comment]
Yep, that's a mystery. Might be a bug in the `ReliableDeliveryActor` responsible for guaranteed delivery of system messages inside Akka.Remote

**Comment by rogeralsing** [comment]
Or is this intended, I found this peice of code:

```csharp
        private void ReceiveHeartbeatRsp(int uid)
        {
            var from = Sender.Path.Address;

            if (_failureDetector.IsMonitoring(from))
            {
                _log.Debug("Received heartbeat rsp from [{0}]", from);
            }
            else
            {
                _log.Debug("Received first heartbeat rsp from [{0}]", from);
            }

            if (_watchingNodes.Contains(from) && !_unreachable.Contains(from))  //<--- checks if we are watching this node
            {
                if (!_addressUids.ContainsKey(from) || _addressUids.ContainsKey(from))
                    ReWatch(from);                                                                                         //<-- sends rewatch
                _addressUids[from] = uid;
                _failureDetector.Heartbeat(from);
            }
        }
```

See inline comments

**Comment by Aaronontheweb** [comment]
Oh yeah, definitely looks intended. Which class is this from? I'd be interested to see if it's still done this way on the JVM.

**Comment by rogeralsing** [comment]
RemoteWatcher.cs

**Comment by rogeralsing** [comment]
In any case, the above code explans why the rewatch appears in sync with the heartbeats.
But it seems expensive to do this for each beat imo.

**Comment by Aaronontheweb** [comment]
@rogeralsing looks like they still do this on the JVM too https://github.com/akka/akka/blob/master/akka-remote/src/main/scala/akka/remote/RemoteWatcher.scala#L150

so is there something we need to uncomment in order to enable this?

As far as the overhead goes, these messages are pretty small - but I can see why they added them from a semantic standpoint (re-affirm upon heartbeat that the remotely watched actors still exist.) Would love to hear what the reasoning was for introducing this though.

**Comment by rogeralsing** [comment]
I sent a new mini PR for this #1091

**Comment by Aaronontheweb** [comment]
Closed by #1099
